File: backend/src/main/java/com/grouply/backend/project/CreateProjectDTO.java
package com.grouply.backend.project;

import com.grouply.backend.project_member.ProjectMember;
import com.grouply.backend.project_member.ProjectPosition;
import com.grouply.backend.technology.Technology;

import lombok.Data;


import java.time.LocalDateTime;

import java.util.Set;

@Data
public class CreateProjectDTO {

    private String name;
    private LocalDateTime createdAt;
    private ProjectStatus status;
    private Set<Technology> technologies;
    private ProjectPosition userPosition;

}

File: frontend/src/features/dashboard/shared/api/CreateProjectDto.ts
import type { TechnologyDTO } from "../../../../shared/models/TechnologyDto"
import type { ProjectPosition } from "../../../../shared/models/project/ProjectPosition"
import type { ProjectStatus } from "../../../../shared/models/project/ProjectStatus"



export interface CreateProjectDTO {
    
    name: string
    githubUrl: string
    status: ProjectStatus
    userPosition: ProjectPosition
    technologies: TechnologyDTO[]
}